### DISCUSSION TITLE
[HELP!] __VUE_PROD_DEVTOOLS__ is not defined

### DISCUSSION THREAD

**Discussion by ipacs13** [nuxt/nuxt#20730] [Questions] [answered]
- Operating System: `Windows_NT`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.0.0-27379487.e0cf2d7`
- Package Manager:  `yarn@1.22.19`
- Bundler:          `Vite`
- User Config:      `vue`, `ssr`, `serverMiddleware`, `meta`, `css`, `buildModules`, `plugins`, `vite`, `app`, `build`
- Runtime Modules:  `-`
- Build Modules:    `unplugin-icons/nuxt`

**Action**
1. `yarn build`
2. `yarn preview`

**Output**

![image](https://user-images.githubusercontent.com/10388785/236889882-caa056cb-043f-446c-9bc0-6dcdfa3b5986.png)

Hello Everyone,

Does anyone of you here experienced this? I've been trying to find out what causes this error when previewing build files. Your response is highly appreciated :)

**Comment by warflash** [comment]
FYI that nuxt version is well over a year old so if it's specific to that version (which it probably is), then it's very unlikely there's still someone around running that and being able to provide insight. I'd recommend upgrading to 3.4.3 to see whether that issue persists

**Reply by ipacs13** [reply]
@warflash Thank you for replying to my questions. I tried upgrading to 3.4.3. However, I face another issues

1. `yarn dev`
![image](https://user-images.githubusercontent.com/10388785/236896937-3ed24ecd-a2fe-43cd-82b9-53b89832d8e1.png)

2. `yarn build`
![image](https://user-images.githubusercontent.com/10388785/236897007-c056c1ab-e55a-4cdd-b8f9-745e1a57b8c5.png)

Do you have an idea what are the causes for this error? Your response is highly appreciated :)

**Reply by warflash** [reply, marked as answer]
Is this in a new project or are you migrating/upgrading an existing one? If it's a new one without much business logic I'd recommend throwing it in a stackblitz so it's easier to have content on things. Since you seem to be getting a lot of errors related to dependencies I'd also recommend clearing your local package manager cache, deleting node_modules as well as deleting and recreating your lockfile. If that doesnt help you could try setting up a brand new nuxt project to see if that works

**Reply by ipacs13** [reply]
Hi @warflash I followed your recommendation and somehow it worked. However, a new issue occurs.

I am seeing this whenever I try to preview the build. 
![image](https://github.com/nuxt/nuxt/assets/10388785/00a5d4f6-0147-42a4-8ce8-e67b50c59929)

Here's my `firebase.client.ts` inside `plugins` folder:

```
import { initializeApp } from "firebase/app";
import { getAuth } from 'firebase/auth';

const firebaseConfig = {
  /** Configuration here.. */
};


import { defineNuxtPlugin } from '#app';


export default defineNuxtPlugin((nuxtApp) => {
  // Initialize Firebase
  const app = initializeApp(firebaseConfig);

  // Get Firebase Authentication instance
  const auth = getAuth(app);
  nuxtApp.$auth = auth;
});
```

I have tried several methods I found in the internet but no luck. Your response is much appreciated :)

**Reply by warflash** [reply]
Nice, glad you got it resolved. Regarding firebase I'd recommend closing the 2 issues you were having as resolved and opening up a new one so people can find it easier/don't get confused. I've never used firebase alongside nuxt as I'm not a huge fan of it but perhaps using https://vuefire.vuejs.org/nuxt/getting-started.html might work instead of spinning up your own plugin?
